### PR TITLE
Alan's pass at cudaizing the 3-d advance routine for the heat

### PR DESCRIPTION
**Pull request by alancalder**
equation test.

### CODE REVIEW

File: Tutorials/Basic/HeatEquation_EX1_C/advance_3d.F90 [hunk 1]
```diff
@@ -11,74 +115,87 @@ subroutine compute_flux (lo, hi, phi, philo, phihi, &
...
+  cuda_result = cudaMemcpyAsync(fzhi_d, fzhi, 3, cudaMemcpyHostToDevice, stream)
+
+  cuda_result = cudaMemcpyAsync(dx_d, dx, 3, cudaMemcpyHostToDevice, stream)
+
+  call threads_and_blocks(lo, hi, numBlocks, numThreads)
+
+  call cuda_compute_flux<<<numBlocks, numThreads, 0, stream>>>(lo_d, hi_d, phi, philo_d, phihi_d, &
+                                                               fluxx, fxlo_d, fxhi_d, &
+                                                               fluxy, fylo_d, fyhi_d, &
+                                                               fluxy, fzlo_d, fzhi_d, dx_d)
```
Comment (zingale) [inline Tutorials/Basic/HeatEquation_EX1_C/advance_3d.F90]: the `fluxy` should be a `fluxz` here

Comment (maxpkatz) [inline Tutorials/Basic/HeatEquation_EX1_C/advance_3d.F90]: This should be a device array

**Comment by zingale** [comment]
looks good

File: Tutorials/Basic/HeatEquation_EX1_C/advance_3d.F90 (hunk 1 above)
Comment (alancalder) [inline Tutorials/Basic/HeatEquation_EX1_C/advance_3d.F90]: Fixed!

**Pull request merged**